### PR TITLE
glue: add GlueCheckpointDB and remove external TiDB usage

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
complete glue to enable integrated into TiDB

### What is changed and how it works?

- add a glue implemented CheckpointDB, which is not elegant since it repeated many MySQLCheckpointDB code. Will refine later or in this PR.
- resome some usage of `TiDBManager`. now `NewTiDBManager` is only called in `tidb-lightning-ctl`
- remove usage of TiDB status port through TLS. now TLS is used only supposed to connect PD/TiKV
- move database name from `USE db` to CREATE TABLE statements, because we may not rely on idle `(sql.DB).ExecContext` use same connection to execute sequentially SQLs, and in future we may support concurently create tables
- remove `taskID` and replace it with `cfg.TaskID`, because `taskID` didn't have another value

### Check List 

Tests 

 - manually test integrated into TiDB and import a sample data from integration test

Side effects

 - Increased code complexity

Related changes

### CODE REVIEW

**Comment by lance6716** [comment]
To design an interface to implemented by both TiDB's data structure (most work) and database/sql, and make `MySQLCheckpointDB` reusable, I found those works are needed:

- to support `(RowInterface).Scan(dst ...interface{})`, in other words to convert a row from TiDB's result set to golang's variables accroding to their types, there's ~260 lines needed https://github.com/golang/go/blob/869e2957b9f66021581b839cadce6cb48ad46114/src/database/sql/convert.go#L219 
- to support `Exec(query string, args ...interface{})`, in other words to interpolate arguments to SQL string, ~100 lines needed https://github.com/go-sql-driver/mysql/blob/3b935426341bc5d229eafd936e4f4240da027ccd/connection.go#L198

lines are significently less than `glue_checkpoint.go`. 

I guess we must repeat some code of `MySQLCheckpointDB` (currently) or repeat some code of go's SQL driver to implement glue checkpoint. Should I switch to another way? @kennytm @glorv @overvenus

**Comment by overvenus** [comment]
For now, I don't think we need to implement 'Scan' and 'Exec'. We can go on with the current implementation.

File: lightning/checkpoints/checkpoints.go [hunk 1]
```diff
@@ -67,6 +67,115 @@ const (
...
+		WHERE (table_name, engine_id, path, offset) = (?, ?, ?, ?);`
+	UpdateTableRebaseTemplate = `
+		UPDATE %s.%s SET alloc_base = GREATEST(?, alloc_base) WHERE table_name = ?;`
+	UpdateTableStatusTemplate = `
+		UPDATE %s.%s SET status = ? WHERE table_name = ?;`
+	UpdateEngineTemplate = `
+		UPDATE %s.%s SET status = ? WHERE (table_name, engine_id) = (?, ?);`
+	DeleteChunkTemplate  = "DELETE FROM %s.%s WHERE table_name = ?;"
+	DeleteEngineTemplate = "DELETE FROM %s.%s WHERE table_name = ?;"
+	DeleteTableTemplate  = "DELETE FROM %s.%s WHERE table_name = ?;"
```
Comment (kennytm) [inline lightning/checkpoints/checkpoints.go:176]: can we use a single constant for all three? 😅

**Review by kennytm — COMMENTED** [review]
Rest LGTM

**Comment by lance6716** [comment]
Add new change:

- move database name from `USE db` to CREATE TABLE statements, because we may not rely on idle `(sql.DB).ExecContext` use same connection to execute sequentially SQLs, and in future we may support concurently create tables

PTAL @kennytm @glorv @overvenus

**Review by kennytm — COMMENTED** [review]
LGTM

**Review by glorv — APPROVED** [review]
LGTM

**Pull request merged**